src/daft-core/src/array/ops/cast.rs

```diff
@@ -598,10 +600,12 @@ macro_rules! pycast_then_arrowcast {
                         .getattr(pyo3::intern!(py, $pytype_str))?
                 };
 
+                let py_dtype = PyDataType { dtype: $daft_type.clone() };
+
                 old_daft_series
                     .call_method1(
                         (pyo3::intern!(py, "_pycast_to_pynative")),
```

Comment:
I wanted to rewrite this function inlined into Rust using https://docs.rs/inline-python/latest/inline_python/, but that requires a newer compiler. Will do later